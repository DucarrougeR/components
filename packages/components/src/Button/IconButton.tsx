import styled from 'styled-components'
import {
  CompatibleHTMLProps,
  psuedoClasses,
  PsuedoProps,
  reset,
  SpaceProps,
  space,
  SizeLarge,
  SizeMedium,
  SizeSmall,
  SizeXSmall,
  SizeXXSmall,
} from '@looker/design-tokens'
import { IconNames } from '@looker/icons'
import React, { forwardRef, Ref } from 'react'
import { Icon } from '../Icon'
import { VisuallyHidden } from '../VisuallyHidden'
import { buttonCSS } from './Button'
import { IconButtonVariantProps, iconButtonVariant } from './variant'

export type IconButtonSizes =
  | SizeXXSmall
  | SizeXSmall
  | SizeSmall
  | SizeMedium
  | SizeLarge

export interface IconButtonProps
  extends Omit<CompatibleHTMLProps<HTMLButtonElement>, 'children' | 'type'>,
    IconButtonVariantProps,
    PsuedoProps,
    SpaceProps {
  type?: 'button' | 'submit' | 'reset'

  /**
   * The Icon to display inside of the button
   */
  icon: IconNames
  /**
   *  A hidden text label for the IconButton that is accessible to assistive technology
   */
  label: string
  /**
   *  Sets the size of the button
   * @default 'xsmall'
   */
  size?: IconButtonSizes
}

const iconSizeHelper = (size: IconButtonSizes) => {
  switch (size) {
    case 'xxsmall':
      return 12
    case 'xsmall':
      return 16
    case 'small':
      return 20
    case 'medium':
      return 28
    case 'large':
    default:
      return 36
  }
}

const IconButtonComponent = forwardRef(
  (props: IconButtonProps, ref: Ref<HTMLButtonElement>) => {
    const { icon, size, label } = props
    return (
      <IconButtonBase ref={ref} color="neutral" p="none" {...props}>
        <VisuallyHidden>{label}</VisuallyHidden>
        <Icon
          name={icon}
          size={iconSizeHelper(size || 'xsmall')}
          aria-hidden={true}
        />
      </IconButtonBase>
    )
  }
)

const IconButtonBase = styled.button<IconButtonProps>`
  ${reset}
  ${space}

  ${buttonCSS}
  padding: 3px;

  ${iconButtonVariant}
  ${psuedoClasses}
`

export const IconButton = styled(IconButtonComponent)``
