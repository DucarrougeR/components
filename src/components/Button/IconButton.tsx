import { SemanticColor, SemanticColors, styled } from '../../style'
import {
  SizeLarge,
  SizeMedium,
  SizeSmall,
  SizeXSmall,
  SizeXXSmall,
} from '../../types'
import { BoxPropsWithout } from '../Box'
import { Button } from '../Button'
import { Icon, IconNames } from '../Icon'
import { VisuallyHidden } from '../VisuallyHidden'

import * as React from 'react'

export type IconButtonSizes =
  | SizeXXSmall
  | SizeXSmall
  | SizeSmall
  | SizeMedium
  | SizeLarge

export interface IconButtonProps
  extends BoxPropsWithout<HTMLButtonElement, 'type' | 'color'> {
  /**
   *  Determines if the icon button has a border
   *  @default false
   */
  outline?: boolean
  /**
   * The Icon to display inside of the button
   */
  icon: IconNames
  /**
   *  A hidden text label for the IconButton that is accessible to assistive technology
   */
  label: string
  /**
   *  Sets the size of the button
   * @default 'xsmall'
   */
  size?: IconButtonSizes
  /**
   * Defines the color of the button. Can be the string name of a color listed in the color theme, or a color object.
   */
  color?: keyof SemanticColors | SemanticColor
}

function iconSizeHelper(size: IconButtonSizes) {
  switch (size) {
    case 'xxsmall':
      return 10
    case 'xsmall':
      return 14
    case 'small':
      return 18
    case 'medium':
      return 26
    default:
      return 33
  }
}

const InternalIconButton: React.FC<IconButtonProps> = ({
  outline,
  icon,
  size,
  label,
  color,
  ...boxProps
}) => (
  <Button
    color={color || 'neutral'}
    type="button"
    variant={outline ? 'outline' : 'transparent'}
    p="xxsmall"
    {...boxProps}
  >
    <VisuallyHidden is="span">{label}</VisuallyHidden>
    <Icon name={icon} size={iconSizeHelper(size || 'xsmall')} />
  </Button>
)

const IconButtonFactory = React.forwardRef((props: IconButtonProps, ref) => (
  <InternalIconButton
    innerRef={ref as React.RefObject<HTMLElement>}
    {...props}
  />
))

export const IconButton = styled(IconButtonFactory)``
