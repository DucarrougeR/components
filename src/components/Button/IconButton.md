`IconButton`'s are used when you need an icon to trigger an action, such as showing a dialog or deleting an item. Icon buttons require a `label` that describes it's action. The `label` should be descriptive and succinct.

`IconButton`'s come in five sizes, `xxsmall`, `xsmall` (default), `small`, `medium`, and `large`.

If you need an icon for purely decorative purposes use an `<Icon />` instead.

```js
import { IconButton } from './IconButton'
import { Tooltip } from '../Tooltip'
import { styled } from '../../style'

const FancyIconButton = styled(IconButton)`
  transition: transform 0.5s;
  &:hover {
    transform: rotate(45deg);
  }
`

;<>
  <IconButton label="Add File" icon="Plus" size="xxsmall" mr="small" />
  <IconButton label="Settings" icon="Gear" mr="small" />
  <IconButton
    label="Add to Favorites"
    icon="Favorite"
    size="small"
    mr="small"
  />
  <IconButton
    label="Delete Look"
    icon="Trash"
    size="medium"
    color="danger"
    mr="small"
  />

  <Tooltip content="Add a new file">
    {(eventHandlers, ref) => (
      <FancyIconButton
        label="Close"
        icon="Close"
        size="large"
        outline
        mr="small"
        innerRef={ref}
        {...eventHandlers}
      />
    )}
  </Tooltip>
</>
```
