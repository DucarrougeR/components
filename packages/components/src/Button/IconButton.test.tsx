import 'jest-styled-components'
import * as React from 'react'
import { assertSnapshot } from '@looker/components-test-utils'
import { IconButton } from './IconButton'

const noop = () => {
  return
}

test('IconButton default', () => {
  assertSnapshot(<IconButton label="Test" icon="Favorite" />)
})

test('IconButton outline', () => {
  assertSnapshot(<IconButton label="Test" icon="Favorite" outline />)
})

test('IconButton resized', () => {
  assertSnapshot(<IconButton label="Test" icon="Favorite" size="large" />)
})

test('IconButton accepts color', () => {
  assertSnapshot(<IconButton label="Test" icon="Favorite" color="danger" />)
})

test('IconButton allows for ARIA attributes', () => {
  assertSnapshot(<IconButton label="Test" icon="Favorite" aria-haspopup />)
})

test('IconButton accepts events', () => {
  assertSnapshot(
    <IconButton label="Test" icon="Favorite" onMouseEnter={noop} />
  )
  assertSnapshot(<IconButton label="Test" icon="Favorite" onClick={noop} />)
})
