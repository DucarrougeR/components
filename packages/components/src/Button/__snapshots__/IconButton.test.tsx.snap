// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconButton accepts color 1`] = `
.c3 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  width: 16px;
  height: 16px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

.c2 {
  position: absolute;
  height: 1px;
  width: 1px;
  overflow: hidden;
  -webkit-clip: rect(1px,1px,1px,1px);
  clip: rect(1px,1px,1px,1px);
}

.c0 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 0.25rem;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-weight: 600;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  outline: none;
  -webkit-transition: border 80ms;
  transition: border 80ms;
  vertical-align: middle;
  white-space: nowrap;
  padding: 0rem;
  height: auto;
}

.c0:focus {
  box-shadow: 0 0 0 0.15rem rgba(204,31,54,0.25);
}

.c0[disabled] {
  cursor: default;
  -webkit-filter: grayscale(0.3);
  filter: grayscale(0.3);
  opacity: 0.25;
}

.c1 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  padding: 3px;
  background: transparent;
  border: 1px solid transparent;
  color: #CC1F36;
}

.c1:focus {
  box-shadow: 0 0 0 0.15rem rgba(204,31,54,0.25);
}

.c1:active,
.c1.active {
  background: #FFE5E9;
  border-color: #FFE5E9;
  color: #ED3B53;
}

.c1:hover,
.c1:focus,
.c1.hover {
  background: #FFF2F4;
  border-color: #FFF2F4;
  color: #ED3B53;
}

.c1[disabled]:hover,
.c1[disabled]:active,
.c1[disabled]:focus {
  background-color: transparent;
  border-color: transparent;
  color: #CC1F36;
}

<button
  className="c0 c1"
  color="danger"
  label="Test"
>
  <div
    className="c2"
  >
    Test
  </div>
  <div
    aria-hidden={true}
    className="c3 "
    size={16}
  >
    <svg
      fill="currentColor"
      height="100%"
      viewBox="0 0 24 24"
      width="100%"
    >
      <title>
        Favorite
      </title>
      <path
        d="M12 21.35l-1.45-1.32C5.4 15.36 2 12.28 2 8.5 2 5.42 4.42 3 7.5 3c1.74 0 3.41.81 4.5 2.09C13.09 3.81 14.76 3 16.5 3 19.58 3 22 5.42 22 8.5c0 3.78-3.4 6.86-8.55 11.54L12 21.35z"
        fill="currentColor"
      />
    </svg>
  </div>
</button>
`;

exports[`IconButton accepts events 1`] = `
.c3 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  width: 16px;
  height: 16px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

.c2 {
  position: absolute;
  height: 1px;
  width: 1px;
  overflow: hidden;
  -webkit-clip: rect(1px,1px,1px,1px);
  clip: rect(1px,1px,1px,1px);
}

.c0 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 0.25rem;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-weight: 600;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  outline: none;
  -webkit-transition: border 80ms;
  transition: border 80ms;
  vertical-align: middle;
  white-space: nowrap;
  padding: 0rem;
  height: auto;
}

.c0:focus {
  box-shadow: 0 0 0 0.15rem rgba(147,155,165,0.25);
}

.c0[disabled] {
  cursor: default;
  -webkit-filter: grayscale(0.3);
  filter: grayscale(0.3);
  opacity: 0.25;
}

.c1 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  padding: 3px;
  background: transparent;
  border: 1px solid transparent;
  color: #64518A;
}

.c1:focus {
  box-shadow: 0 0 0 0.15rem rgba(100,81,138,0.25);
}

.c1:active,
.c1.active {
  background: #E8E5FF;
  border-color: #E8E5FF;
  color: #6C43E0;
}

.c1:hover,
.c1:focus,
.c1.hover {
  background: #F3F2FF;
  border-color: #F3F2FF;
  color: #6C43E0;
}

.c1[disabled]:hover,
.c1[disabled]:active,
.c1[disabled]:focus {
  background-color: transparent;
  border-color: transparent;
  color: #64518A;
}

<button
  className="c0 c1"
  color="neutral"
  label="Test"
  onMouseEnter={[Function]}
>
  <div
    className="c2"
  >
    Test
  </div>
  <div
    aria-hidden={true}
    className="c3 "
    size={16}
  >
    <svg
      fill="currentColor"
      height="100%"
      viewBox="0 0 24 24"
      width="100%"
    >
      <title>
        Favorite
      </title>
      <path
        d="M12 21.35l-1.45-1.32C5.4 15.36 2 12.28 2 8.5 2 5.42 4.42 3 7.5 3c1.74 0 3.41.81 4.5 2.09C13.09 3.81 14.76 3 16.5 3 19.58 3 22 5.42 22 8.5c0 3.78-3.4 6.86-8.55 11.54L12 21.35z"
        fill="currentColor"
      />
    </svg>
  </div>
</button>
`;

exports[`IconButton accepts events 2`] = `
.c3 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  width: 16px;
  height: 16px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

.c2 {
  position: absolute;
  height: 1px;
  width: 1px;
  overflow: hidden;
  -webkit-clip: rect(1px,1px,1px,1px);
  clip: rect(1px,1px,1px,1px);
}

.c0 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 0.25rem;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-weight: 600;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  outline: none;
  -webkit-transition: border 80ms;
  transition: border 80ms;
  vertical-align: middle;
  white-space: nowrap;
  padding: 0rem;
  height: auto;
}

.c0:focus {
  box-shadow: 0 0 0 0.15rem rgba(147,155,165,0.25);
}

.c0[disabled] {
  cursor: default;
  -webkit-filter: grayscale(0.3);
  filter: grayscale(0.3);
  opacity: 0.25;
}

.c1 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  padding: 3px;
  background: transparent;
  border: 1px solid transparent;
  color: #64518A;
}

.c1:focus {
  box-shadow: 0 0 0 0.15rem rgba(100,81,138,0.25);
}

.c1:active,
.c1.active {
  background: #E8E5FF;
  border-color: #E8E5FF;
  color: #6C43E0;
}

.c1:hover,
.c1:focus,
.c1.hover {
  background: #F3F2FF;
  border-color: #F3F2FF;
  color: #6C43E0;
}

.c1[disabled]:hover,
.c1[disabled]:active,
.c1[disabled]:focus {
  background-color: transparent;
  border-color: transparent;
  color: #64518A;
}

<button
  className="c0 c1"
  color="neutral"
  label="Test"
  onClick={[Function]}
>
  <div
    className="c2"
  >
    Test
  </div>
  <div
    aria-hidden={true}
    className="c3 "
    size={16}
  >
    <svg
      fill="currentColor"
      height="100%"
      viewBox="0 0 24 24"
      width="100%"
    >
      <title>
        Favorite
      </title>
      <path
        d="M12 21.35l-1.45-1.32C5.4 15.36 2 12.28 2 8.5 2 5.42 4.42 3 7.5 3c1.74 0 3.41.81 4.5 2.09C13.09 3.81 14.76 3 16.5 3 19.58 3 22 5.42 22 8.5c0 3.78-3.4 6.86-8.55 11.54L12 21.35z"
        fill="currentColor"
      />
    </svg>
  </div>
</button>
`;

exports[`IconButton allows for ARIA attributes 1`] = `
.c3 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  width: 16px;
  height: 16px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

.c2 {
  position: absolute;
  height: 1px;
  width: 1px;
  overflow: hidden;
  -webkit-clip: rect(1px,1px,1px,1px);
  clip: rect(1px,1px,1px,1px);
}

.c0 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 0.25rem;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-weight: 600;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  outline: none;
  -webkit-transition: border 80ms;
  transition: border 80ms;
  vertical-align: middle;
  white-space: nowrap;
  padding: 0rem;
  height: auto;
}

.c0:focus {
  box-shadow: 0 0 0 0.15rem rgba(147,155,165,0.25);
}

.c0[disabled] {
  cursor: default;
  -webkit-filter: grayscale(0.3);
  filter: grayscale(0.3);
  opacity: 0.25;
}

.c1 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  padding: 3px;
  background: transparent;
  border: 1px solid transparent;
  color: #64518A;
}

.c1:focus {
  box-shadow: 0 0 0 0.15rem rgba(100,81,138,0.25);
}

.c1:active,
.c1.active {
  background: #E8E5FF;
  border-color: #E8E5FF;
  color: #6C43E0;
}

.c1:hover,
.c1:focus,
.c1.hover {
  background: #F3F2FF;
  border-color: #F3F2FF;
  color: #6C43E0;
}

.c1[disabled]:hover,
.c1[disabled]:active,
.c1[disabled]:focus {
  background-color: transparent;
  border-color: transparent;
  color: #64518A;
}

<button
  aria-haspopup={true}
  className="c0 c1"
  color="neutral"
  label="Test"
>
  <div
    className="c2"
  >
    Test
  </div>
  <div
    aria-hidden={true}
    className="c3 "
    size={16}
  >
    <svg
      fill="currentColor"
      height="100%"
      viewBox="0 0 24 24"
      width="100%"
    >
      <title>
        Favorite
      </title>
      <path
        d="M12 21.35l-1.45-1.32C5.4 15.36 2 12.28 2 8.5 2 5.42 4.42 3 7.5 3c1.74 0 3.41.81 4.5 2.09C13.09 3.81 14.76 3 16.5 3 19.58 3 22 5.42 22 8.5c0 3.78-3.4 6.86-8.55 11.54L12 21.35z"
        fill="currentColor"
      />
    </svg>
  </div>
</button>
`;

exports[`IconButton default 1`] = `
.c3 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  width: 16px;
  height: 16px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

.c2 {
  position: absolute;
  height: 1px;
  width: 1px;
  overflow: hidden;
  -webkit-clip: rect(1px,1px,1px,1px);
  clip: rect(1px,1px,1px,1px);
}

.c0 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 0.25rem;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-weight: 600;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  outline: none;
  -webkit-transition: border 80ms;
  transition: border 80ms;
  vertical-align: middle;
  white-space: nowrap;
  padding: 0rem;
  height: auto;
}

.c0:focus {
  box-shadow: 0 0 0 0.15rem rgba(147,155,165,0.25);
}

.c0[disabled] {
  cursor: default;
  -webkit-filter: grayscale(0.3);
  filter: grayscale(0.3);
  opacity: 0.25;
}

.c1 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  padding: 3px;
  background: transparent;
  border: 1px solid transparent;
  color: #64518A;
}

.c1:focus {
  box-shadow: 0 0 0 0.15rem rgba(100,81,138,0.25);
}

.c1:active,
.c1.active {
  background: #E8E5FF;
  border-color: #E8E5FF;
  color: #6C43E0;
}

.c1:hover,
.c1:focus,
.c1.hover {
  background: #F3F2FF;
  border-color: #F3F2FF;
  color: #6C43E0;
}

.c1[disabled]:hover,
.c1[disabled]:active,
.c1[disabled]:focus {
  background-color: transparent;
  border-color: transparent;
  color: #64518A;
}

<button
  className="c0 c1"
  color="neutral"
  label="Test"
>
  <div
    className="c2"
  >
    Test
  </div>
  <div
    aria-hidden={true}
    className="c3 "
    size={16}
  >
    <svg
      fill="currentColor"
      height="100%"
      viewBox="0 0 24 24"
      width="100%"
    >
      <title>
        Favorite
      </title>
      <path
        d="M12 21.35l-1.45-1.32C5.4 15.36 2 12.28 2 8.5 2 5.42 4.42 3 7.5 3c1.74 0 3.41.81 4.5 2.09C13.09 3.81 14.76 3 16.5 3 19.58 3 22 5.42 22 8.5c0 3.78-3.4 6.86-8.55 11.54L12 21.35z"
        fill="currentColor"
      />
    </svg>
  </div>
</button>
`;

exports[`IconButton outline 1`] = `
.c3 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  width: 16px;
  height: 16px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

.c2 {
  position: absolute;
  height: 1px;
  width: 1px;
  overflow: hidden;
  -webkit-clip: rect(1px,1px,1px,1px);
  clip: rect(1px,1px,1px,1px);
}

.c0 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 0.25rem;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-weight: 600;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  outline: none;
  -webkit-transition: border 80ms;
  transition: border 80ms;
  vertical-align: middle;
  white-space: nowrap;
  padding: 0rem;
  height: auto;
}

.c0:focus {
  box-shadow: 0 0 0 0.15rem rgba(147,155,165,0.25);
}

.c0[disabled] {
  cursor: default;
  -webkit-filter: grayscale(0.3);
  filter: grayscale(0.3);
  opacity: 0.25;
}

.c1 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  padding: 3px;
  background: #FFFFFF;
  border: 1px solid #C1C6CC;
  color: #64518A;
}

.c1:focus {
  box-shadow: 0 0 0 0.15rem rgba(100,81,138,0.25);
}

.c1:active,
.c1.active {
  background: #64518A;
  border-color: #64518A;
  color: #FFFFFF;
}

.c1:hover,
.c1:focus,
.c1.hover {
  background: #FFFFFF;
  border-color: #64518A;
  color: #2A1B60;
}

.c1[disabled]:hover,
.c1[disabled]:active,
.c1[disabled]:focus {
  background-color: #FFFFFF;
  border-color: #C1C6CC;
  color: #64518A;
}

<button
  className="c0 c1"
  color="neutral"
  label="Test"
>
  <div
    className="c2"
  >
    Test
  </div>
  <div
    aria-hidden={true}
    className="c3 "
    size={16}
  >
    <svg
      fill="currentColor"
      height="100%"
      viewBox="0 0 24 24"
      width="100%"
    >
      <title>
        Favorite
      </title>
      <path
        d="M12 21.35l-1.45-1.32C5.4 15.36 2 12.28 2 8.5 2 5.42 4.42 3 7.5 3c1.74 0 3.41.81 4.5 2.09C13.09 3.81 14.76 3 16.5 3 19.58 3 22 5.42 22 8.5c0 3.78-3.4 6.86-8.55 11.54L12 21.35z"
        fill="currentColor"
      />
    </svg>
  </div>
</button>
`;

exports[`IconButton resized 1`] = `
.c3 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  width: 36px;
  height: 36px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

.c2 {
  position: absolute;
  height: 1px;
  width: 1px;
  overflow: hidden;
  -webkit-clip: rect(1px,1px,1px,1px);
  clip: rect(1px,1px,1px,1px);
}

.c0 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 0.25rem;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-weight: 600;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  outline: none;
  -webkit-transition: border 80ms;
  transition: border 80ms;
  vertical-align: middle;
  white-space: nowrap;
  font-size: 1.375rem;
  height: 42px;
  padding-left: 1.25rem;
  padding-right: 1.25rem;
  padding: 0rem;
  height: auto;
}

.c0:focus {
  box-shadow: 0 0 0 0.15rem rgba(147,155,165,0.25);
}

.c0[disabled] {
  cursor: default;
  -webkit-filter: grayscale(0.3);
  filter: grayscale(0.3);
  opacity: 0.25;
}

.c1 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  padding: 3px;
  background: transparent;
  border: 1px solid transparent;
  color: #64518A;
}

.c1:focus {
  box-shadow: 0 0 0 0.15rem rgba(100,81,138,0.25);
}

.c1:active,
.c1.active {
  background: #E8E5FF;
  border-color: #E8E5FF;
  color: #6C43E0;
}

.c1:hover,
.c1:focus,
.c1.hover {
  background: #F3F2FF;
  border-color: #F3F2FF;
  color: #6C43E0;
}

.c1[disabled]:hover,
.c1[disabled]:active,
.c1[disabled]:focus {
  background-color: transparent;
  border-color: transparent;
  color: #64518A;
}

<button
  className="c0 c1"
  color="neutral"
  label="Test"
  size="large"
>
  <div
    className="c2"
  >
    Test
  </div>
  <div
    aria-hidden={true}
    className="c3 "
    size={36}
  >
    <svg
      fill="currentColor"
      height="100%"
      viewBox="0 0 24 24"
      width="100%"
    >
      <title>
        Favorite
      </title>
      <path
        d="M12 21.35l-1.45-1.32C5.4 15.36 2 12.28 2 8.5 2 5.42 4.42 3 7.5 3c1.74 0 3.41.81 4.5 2.09C13.09 3.81 14.76 3 16.5 3 19.58 3 22 5.42 22 8.5c0 3.78-3.4 6.86-8.55 11.54L12 21.35z"
        fill="currentColor"
      />
    </svg>
  </div>
</button>
`;
